File: easy/349_intersection_of_two_arrays.rb
# 349. Intersection of Two Arrays
# Easy
#
# Given two integer arrays nums1 and nums2, return an array of their
# intersection. Each element in the result must be unique and you may return
# the result in any order.

def intersection(nums1, nums2)
  h = {}
  nums1.each do |n|
    h[n] = true
  end

  res = []
  nums2.uniq.each do |n|
    res << n if h.has_key?(n)
  end

  res
end

File: python/349_intersection_of_two_arrays.py
"""
# 349. Intersection of Two Arrays

Easy

Given two integer arrays `nums1` and `nums2`, return an array of their intersection. Each element in the result must be unique and you may return the result in any order.
"""


class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        d = set()
        for num in set(nums1):
            d.add(num)

        res = []
        for num in set(nums2):
            if num in d:
                res.append(num)

        return res
